android/source/VideoLocker/src/org/edx/mobile/view/LoginActivity.java

```diff
@@ -573,7 +574,7 @@ public void onFinish(ProfileModel result) {
 
         @Override
         public void onException(Exception ex) {
-            onUserLoginIncorrect(false);
+            onUserLoginIncorrect(false, ex);
```

Comment:
Then this would be responsible for parsing out the LoginErrorMessage. That way the video code doesn't need to know about particular details of the control flow, like the fact that we use exceptions to track errors.